### PR TITLE
[UnitTest] Test if unrolling grouped conv is disabled for Interpreter and CPU backends.

### PR DESCRIPTION
**Pull request by beicy**
This opt is now disabled for Interpreter and CPU backends, and enabled for OpenCL backend.

### CODE REVIEW

File: tests/unittests/graphTest.cpp [hunk 1]
```diff
@@ -299,3 +299,40 @@ TEST(Graph, nodesWithPredicates) {
...
+    EXPECT_EQ(count, 1);
+
+  return count;
+}
+
+// Check the unrolling grouped convolution opt status:
+// -- disabled for Interpreter and CPU backend,
+// -- enabled for openCL backend.
+TEST(Graph, disableUnrollingGroupConv) {
+  int cpu_n = getConvNodeSize(BackendKind::CPU);
```
Comment (rdzhabarov) [inline tests/unittests/graphTest.cpp]: we do not use, '_' in names for c++ code (numberOfNodesCPU, or something like that).
Also, unsigned instead.

File: tests/unittests/graphTest.cpp [hunk 2]
```diff
@@ -299,3 +299,40 @@ TEST(Graph, nodesWithPredicates) {
...
+// Return the number of ConvolutionNode after lower.
+int getConvNodeSize(BackendKind kind) {
+  Module mod;
+  Function *F = mod.createFunction("main");
+  IRFunction M(F);
+  auto *input = mod.createVariable(ElemKind::FloatTy, {1, 2, 1, 32}, "input");
+  ConvolutionNode *CN = F->createConv("conv", input, 6, 1, 1, 0, 2);
+  F->createSave("save", CN);
+
+  std::unique_ptr<Backend> backend_;
```
Comment (rdzhabarov) [inline tests/unittests/graphTest.cpp]: you can combine declaration and init, backend(createBackend).

Comment (rdzhabarov) [reply tests/unittests/graphTest.cpp]: also, no '_', it's only for private members.

File: tests/unittests/graphTest.cpp [hunk 3]
```diff
@@ -299,3 +299,40 @@ TEST(Graph, nodesWithPredicates) {
...
+  backend_.reset(createBackend(kind, &M));
+  lower(F, CompilationMode::Infer, *backend_);
+
+  unsigned count = 0;
+  for (auto *n : F->getNodes()) {
```
Comment (rdzhabarov) [inline tests/unittests/graphTest.cpp]: iirc, there was a method we use in tests to count a number of nodes of a given type. I'd reuse that.

File: tests/unittests/graphTest.cpp [hunk 4]
```diff
@@ -299,3 +299,40 @@ TEST(Graph, nodesWithPredicates) {
...
+    if (n->getKind() == Kinded::Kind::ConvolutionNodeKind) {
+      count++;
+    }
+  }
+
+  if (kind == BackendKind::CPU)
+    EXPECT_EQ(count, 1);
```
Comment (rdzhabarov) [inline tests/unittests/graphTest.cpp]: would be good to use {} consistently. Like having {} around Expect_eq, similar to line 318.

File: tests/unittests/graphTest.cpp (hunk 3 above)
Comment (beicy) [reply tests/unittests/graphTest.cpp]: There is only a static function "static unsigned countNodeKind" in graphOptzTest.cpp.

File: tests/unittests/graphTest.cpp [hunk 5]
```diff
@@ -299,3 +299,40 @@ TEST(Graph, nodesWithPredicates) {
   EE.compile(CompilationMode::Infer, F);
   EE.run({input}, {&inputs});
 }
+
+// Return the number of ConvolutionNode after lower.
+int getConvNodeSize(BackendKind kind) {
```
Comment (rdzhabarov) [inline tests/unittests/graphTest.cpp]: unsigned.

File: tests/unittests/graphTest.cpp [hunk 6]
```diff
@@ -299,3 +299,40 @@ TEST(Graph, nodesWithPredicates) {
...
+  return count;
+}
+
+// Check the unrolling grouped convolution opt status:
+// -- disabled for Interpreter and CPU backend,
+// -- enabled for openCL backend.
+TEST(Graph, disableUnrollingGroupConv) {
+  unsigned numberOfNodesCPU = getConvNodeSize(BackendKind::CPU);
+  unsigned numberOfNodesInterpreter = getConvNodeSize(BackendKind::Interpreter);
+  unsigned numberOfNodesOpenCL = getConvNodeSize(BackendKind::OpenCL);
```
Comment (rdzhabarov) [inline tests/unittests/graphTest.cpp]: as talked offline this only needs to be enabled in case we compiled with OPENCL

**Pull request merged**